From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: SupeChicken666 <me@supechicken666.dev>
Date: Fri, 17 Jul 2026 11:30:33 +0800
Subject: [PATCH 29/33] ueventd: Change ownership of /dev/input to
 system:system

Otherwise our hwcomposer won't be able to create input pipes under /dev/input

Signed-off-by: SupeChicken666 <me@supechicken666.dev>
---
 init/devices.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/init/devices.cpp b/init/devices.cpp
index a119f00..7ec33ff 100644
--- a/init/devices.cpp
+++ b/init/devices.cpp
@@ -829,6 +829,7 @@ void DeviceHandler::HandleUevent(const Uevent& uevent) {
     }
 
     mkdir_recursive(Dirname(devpath), 0755);
+    if (input_dev) chown("/dev/input", AID_SYSTEM, AID_SYSTEM);
 
     HandleDevice(uevent.action, devpath, block, uevent.major, uevent.minor, links);
 
